From f976eba0ddc8bc97ab3049a6a2e4f87110ada029 Mon Sep 17 00:00:00 2001
From: Samuli Piippo <samuli.piippo@qt.io>
Date: Fri, 6 Oct 2017 15:54:35 +0300
Subject: [PATCH] Enable atmel mxt multitouch controller for ixora v1.1 dtb

Need to disable fusion touch to get the new atmel mxt multitouch
controller to work.
---
 arch/arm/boot/dts/imx6q-apalis-ixora-v1.1.dts | 3 ++-
 arch/arm/boot/dts/imx6q-apalis-ixora-v1.2.dts | 3 ++-
 arch/arm/boot/dts/imx6q-apalis-ixora.dts      | 3 ++-
 3 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/arch/arm/boot/dts/imx6q-apalis-ixora-v1.1.dts b/arch/arm/boot/dts/imx6q-apalis-ixora-v1.1.dts
index 6c4aa68999a9..6ce362066c3b 100644
--- a/arch/arm/boot/dts/imx6q-apalis-ixora-v1.1.dts
+++ b/arch/arm/boot/dts/imx6q-apalis-ixora-v1.1.dts
@@ -243,7 +243,7 @@
 		interrupt-parent = <&gpio6>;
 		interrupts = <10 IRQ_TYPE_EDGE_FALLING>;
 		reset-gpios = <&gpio6 9 GPIO_ACTIVE_HIGH>; /* SODIMM 13 */
-		status = "disabled";
+		status = "okay";
 	};
 
 	pcap@10 {
@@ -253,6 +253,7 @@
 		gpios = <&gpio6 10 GPIO_ACTIVE_HIGH /* MXM3 11, Pen down interrupt */
 			 &gpio6  9 GPIO_ACTIVE_HIGH /* MXM3 13, Reset */
 			>;
+		status = "disabled";
 	};
 
 	/* M41T0M6 real time clock on carrier board */
diff --git a/arch/arm/boot/dts/imx6q-apalis-ixora-v1.2.dts b/arch/arm/boot/dts/imx6q-apalis-ixora-v1.2.dts
index c6ffe7477650..cfb6011d7b5c 100644
--- a/arch/arm/boot/dts/imx6q-apalis-ixora-v1.2.dts
+++ b/arch/arm/boot/dts/imx6q-apalis-ixora-v1.2.dts
@@ -259,7 +259,7 @@
 		interrupt-parent = <&gpio6>;
 		interrupts = <10 IRQ_TYPE_EDGE_FALLING>;
 		reset-gpios = <&gpio6 9 GPIO_ACTIVE_HIGH>; /* SODIMM 13 */
-		status = "disabled";
+		status = "okay";
 	};
 
 	pcap@10 {
@@ -269,6 +269,7 @@
 		gpios = <&gpio6 10 GPIO_ACTIVE_HIGH /* MXM3 11, Pen down interrupt */
 			 &gpio6  9 GPIO_ACTIVE_HIGH /* MXM3 13, Reset */
 			>;
+		status = "disabled";
 	};
 
 	/* M41T0M6 real time clock on carrier board */
diff --git a/arch/arm/boot/dts/imx6q-apalis-ixora.dts b/arch/arm/boot/dts/imx6q-apalis-ixora.dts
index 755ea18dcc14..d9947bff9bbe 100644
--- a/arch/arm/boot/dts/imx6q-apalis-ixora.dts
+++ b/arch/arm/boot/dts/imx6q-apalis-ixora.dts
@@ -234,7 +234,7 @@
 		interrupt-parent = <&gpio6>;
 		interrupts = <10 IRQ_TYPE_EDGE_FALLING>;
 		reset-gpios = <&gpio6 9 GPIO_ACTIVE_HIGH>; /* SODIMM 13 */
-		status = "disabled";
+		status = "okay";
 	};
 
 	pcap@10 {
@@ -244,6 +244,7 @@
 		gpios = <&gpio6 10 GPIO_ACTIVE_HIGH /* MXM3 11, Pen down interrupt */
 			 &gpio6  9 GPIO_ACTIVE_HIGH /* MXM3 13, Reset */
 			>;
+		status = "disabled";
 	};
 
 	eeprom@50 {
